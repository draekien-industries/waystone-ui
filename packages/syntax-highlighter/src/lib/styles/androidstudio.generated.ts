export const androidstudio = {
  're code.hljs': {
    'display': 'block',
    'overflowX': 'auto',
    'padding': '1em'
  },
  'ode.hljs': {
    'padding': '3px 5px'
  },
  'hljs': {
    'color': '#a9b7c6',
    'background': '#282b2e'
  },
  'hljs-bullet': {
    'color': '#6897bb'
  },
  'hljs-literal': {
    'color': '#6897bb'
  },
  'hljs-number': {
    'color': '#6897bb'
  },
  'hljs-symbol': {
    'color': '#6897bb'
  },
  'hljs-deletion': {
    'color': '#cc7832'
  },
  'hljs-keyword': {
    'color': '#cc7832'
  },
  'hljs-selector-tag': {
    'color': '#cc7832'
  },
  'hljs-link': {
    'color': '#629755'
  },
  'hljs-template-variable': {
    'color': '#629755'
  },
  'hljs-variable': {
    'color': '#629755'
  },
  'hljs-comment': {
    'color': 'grey'
  },
  'hljs-quote': {
    'color': 'grey'
  },
  'hljs-meta': {
    'color': '#bbb529'
  },
  'hljs-addition': {
    'color': '#6a8759'
  },
  'hljs-attribute': {
    'color': '#6a8759'
  },
  'hljs-string': {
    'color': '#6a8759'
  },
  'hljs-section': {
    'color': '#ffc66d'
  },
  'hljs-title': {
    'color': '#ffc66d'
  },
  'hljs-type': {
    'color': '#ffc66d'
  },
  'hljs-name': {
    'color': '#e8bf6a'
  },
  'hljs-selector-class': {
    'color': '#e8bf6a'
  },
  'hljs-selector-id': {
    'color': '#e8bf6a'
  },
  'hljs-emphasis': {
    'fontStyle': 'italic'
  },
  'hljs-strong': {
    'fontWeight': '700'
  }
}
