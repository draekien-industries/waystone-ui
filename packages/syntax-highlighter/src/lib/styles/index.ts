export * from './a11yDark.generated';
export * from './a11yLight.generated';
export * from './agate.generated';
export * from './anOldHope.generated';
export * from './androidstudio.generated';
export * from './arduinoLight.generated';
export * from './arta.generated';
export * from './ascetic.generated';
export * from './atomOneDarkReasonable.generated';
export * from './atomOneDark.generated';
export * from './atomOneLight.generated';
export * from './brownPaper.generated';
export * from './codepenEmbed.generated';
export * from './colorBrewer.generated';
export * from './dark.generated';
export * from './defaultStyle.generated';
export * from './devibeans.generated';
export * from './docco.generated';
export * from './far.generated';
export * from './felipec.generated';
export * from './foundation.generated';
export * from './githubDarkDimmed.generated';
export * from './githubDark.generated';
export * from './github.generated';
export * from './gml.generated';
export * from './googlecode.generated';
export * from './gradientDark.generated';
export * from './gradientLight.generated';
export * from './grayscale.generated';
export * from './hybrid.generated';
export * from './idea.generated';
export * from './intellijLight.generated';
export * from './irBlack.generated';
export * from './isblEditorDark.generated';
export * from './isblEditorLight.generated';
export * from './kimbieDark.generated';
export * from './kimbieLight.generated';
export * from './lightfair.generated';
export * from './lioshi.generated';
export * from './magula.generated';
export * from './monoBlue.generated';
export * from './monokaiSublime.generated';
export * from './monokai.generated';
export * from './nightOwl.generated';
export * from './nnfxDark.generated';
export * from './nnfxLight.generated';
export * from './nord.generated';
export * from './obsidian.generated';
export * from './pandaSyntaxDark.generated';
export * from './pandaSyntaxLight.generated';
export * from './paraisoDark.generated';
export * from './paraisoLight.generated';
export * from './pojoaque.generated';
export * from './purebasic.generated';
export * from './qtcreatorDark.generated';
export * from './qtcreatorLight.generated';
export * from './rainbow.generated';
export * from './routeros.generated';
export * from './schoolBook.generated';
export * from './shadesOfPurple.generated';
export * from './srcery.generated';
export * from './stackoverflowDark.generated';
export * from './stackoverflowLight.generated';
export * from './sunburst.generated';
export * from './tokyoNightDark.generated';
export * from './tokyoNightLight.generated';
export * from './tomorrowNightBlue.generated';
export * from './tomorrowNightBright.generated';
export * from './vs.generated';
export * from './vs2015.generated';
export * from './xcode.generated';
export * from './xt256.generated';
