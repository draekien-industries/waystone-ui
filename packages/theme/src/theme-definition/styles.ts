export const styles = {
  shadows: {
    none: 'none',
    sm: 'inset 0 0 0.5px 1px hsla(0, 0%, 100%, 0.075), 0 0 0 1px hsla(0, 0%, 0%, 0.05), 0 0.3px 0.4px hsla(0, 0%, 0%, 0.02), 0 0.9px 1.5px hsla(0, 0%, 0%, 0.045), 0 3.5px 6px hsla(0, 0%, 0%, 0.09)',
    md: 'inset 0 0 0.5px 1px hsla(0, 0%, 100%, 0.075), 0 0 4px 1px hsla(0, 0%, 0%, 0.05), 0 0.3px 0.4px hsla(0, 0%, 0%, 0.02), 0 0.9px 1.5px hsla(0, 0%, 0%, 0.045), 0 3.5px 6px hsla(0, 0%, 0%, 0.09)',
    lg: 'inset 0 0 0.5px 1px hsla(0, 0%, 100%, 0.075), 0 0 8px 1px hsla(0, 0%, 0%, 0.05), 0 0.3px 0.4px hsla(0, 0%, 0%, 0.02), 0 0.9px 1.5px hsla(0, 0%, 0%, 0.045), 0 3.5px 6px hsla(0, 0%, 0%, 0.09)',
  },
  sizes: {
    xs: '1em',
    sm: '5em',
    md: '10em',
    lg: '15em',
    xl: '20em',
    half: '50%',
    full: '100%',
  },
} as const;

export default styles;
