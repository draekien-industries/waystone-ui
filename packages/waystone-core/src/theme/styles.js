export const styles = {
    shadows: {
        sm: '0 0 0.25em 0 rgba(4, 40, 52, 0.1), 0 0.125em 0.25em -0.125em rgba(4, 40, 52, 0.25)',
        md: '0 0 0.25em 0 rgba(4, 40, 52, 0.1), 0 0.5em 1em -0.5em rgba(4, 40, 52, 0.25)',
        lg: '0 0 0.5em 0 rgba(4, 40, 52, 0.1), 0 1em 2em -1em rgba(3, 36, 48, 0.25)',
        'sm-dark': '0 0 0.25em 0 #1b1b1bFF, 0 0.125em 0.25em -0.125em #0b0b0bA0',
        'md-dark': '0 0 0.25em 0 #1b1b1bFF, 0 0.5em 1em -0.5em #0b0b0bA0',
        'lg-dark': '0 0 0.5em 0 #1b1b1bFF, 0 1em 2em -1em #0b0b0bA0',
    },
    sizes: {
        xs: '1em',
        sm: '5em',
        md: '10em',
        lg: '15em',
        xl: '20em',
        half: '50%',
        full: '100%',
    },
};
export default styles;
