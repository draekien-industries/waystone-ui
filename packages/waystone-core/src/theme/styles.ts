export const styles = {
  shadows: {
    sm: '0 0 0.25em 0 rgba(4, 40, 52, 0.1), 0 0.125em 0.25em -0.125em rgba(4, 40, 52, 0.25)',
    md: '0 0 0.25em 0 rgba(4, 40, 52, 0.1), 0 0.5em 1em -0.5em rgba(4, 40, 52, 0.25)',
    lg: '0 0 0.5em 0 rgba(4, 40, 52, 0.1), 0 1em 2em -1em rgba(3, 36, 48, 0.25)',
    'sm-dark':
      '0 0 0.5em 0 rgba(27, 27, 27, 1), 0 0.5 1em -0.5em rgba(11, 11, 11, 0.73)',
    'md-dark':
      '0 0 1em 0 rgba(27, 27, 27, 1), 0 1em 2em -1em rgba(11, 11, 11, 0.73)',
    'lg-dark':
      '0 0 2em 0 rgba(27, 27, 27, 1), 0 2em 4em -2em rgba(11, 11, 11, 0.73)',
  },
  sizes: {
    xs: '1em',
    sm: '5em',
    md: '10em',
    lg: '15em',
    xl: '20em',
    half: '50%',
    full: '100%',
  },
};

export default styles;
