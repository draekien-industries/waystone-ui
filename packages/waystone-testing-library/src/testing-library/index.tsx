'use client';

import type { RenderOptions, RenderResult } from '@testing-library/react';
import { render } from '@testing-library/react';
import { Waystone } from '@waystone/theme-provider';
import type { PropsWithChildren } from 'react';
import type React from 'react';
import { useEffect } from 'react';
import { useColorMode } from 'theme-ui';

const TestProvider = ({ children }: PropsWithChildren) => (
  <Waystone>{children}</Waystone>
);

const DarkModeWrapper = ({ children }: PropsWithChildren) => {
  const [, setColorMode] = useColorMode();

  useEffect(() => {
    setColorMode('dark');
  }, [setColorMode]);

  return <div>{children}</div>;
};

const themedRender: (
  ui: React.ReactElement,
  options?: RenderOptions
) => RenderResult = (ui, options?) =>
  render(ui, { wrapper: options?.wrapper || TestProvider, ...options });

export * from '@testing-library/react';
export * from '@testing-library/user-event';

export { DarkModeWrapper, themedRender as render };
