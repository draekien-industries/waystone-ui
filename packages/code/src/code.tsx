import type { PropsWithChildren } from 'react';
import { codeCss } from './code.styles';

export const Code = ({ children }: PropsWithChildren) => (
  <code sx={codeCss}>{children}</code>
);
