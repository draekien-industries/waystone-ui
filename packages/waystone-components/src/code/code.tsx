'use client';

import { ReactNode } from 'react';
import { useColorMode } from 'theme-ui';
import { codeCss } from './code.styles';

/** The props required to render the code component. */
export interface CodeProps {
  /** The code to render. */
  children: ReactNode;
}

/**
 * An inline code snippet.
 */
export const Code = ({ children }: CodeProps) => {
  const [mode] = useColorMode();

  return <code sx={codeCss(mode === 'dark')}>{children}</code>;
};
