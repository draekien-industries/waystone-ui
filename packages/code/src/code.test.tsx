import { render } from '@waystone/testing-library';
import { codeCss } from './code.styles';
import { Code } from './code';

describe('codeCss', () => {
  it('should match snapshot', () => {
    expect(codeCss).toMatchSnapshot();
  });
});

describe('<Code />', () => {
  it('should match snapshot', () => {
    const { container } = render(<Code>{`console.log('Hello World')`}</Code>);

    expect(container).toMatchSnapshot();
  });
});
