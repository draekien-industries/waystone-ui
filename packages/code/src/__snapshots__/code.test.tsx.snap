// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Code /> should match snapshot 1`] = `
<div>
  <code
    class="css-h1jkmd"
  >
    console.log('Hello World')
  </code>
</div>
`;

exports[`codeCss should match snapshot 1`] = `
{
  "backgroundColor": "ghost",
  "borderRadius": "small",
  "color": "code",
  "fontSize": "1em",
  "fontWeight": "medium",
  "paddingX": "extra-small",
  "paddingY": 2,
  "variant": "text.mono",
}
`;
